pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -779,7 +779,8 @@ private void handleOwnEvent(String serviceUnit, ServiceUnitStateData data) {
             lastOwnEventHandledAt = System.currentTimeMillis();
             stateChangeListeners.notify(serviceUnit, data, null);
             log(null, serviceUnit, data, null);
-        } else if ((data.force() || isTransferCommand(data)) && isTargetBroker(data.sourceBroker())) {
+        } else if ((data.force() || isTransferCommand(data)) && isTargetBroker(data.sourceBroker())
+                && pulsar.getConfig().isLoadBalancerMultiPhaseBundleUnload()) {
```

Comment:
we still need to run this block if `data.force() && isTargetBroker(data.sourceBroker()`)

Reply:
I think the logic should be like the following.

if(
isTargetBroker(data.sourceBroker())
&& (data.force()  // from orphan cleanup
          || (pulsar.getConfig().isLoadBalancerMultiPhaseBundleUnload() && isTransferCommand(data)) // from transfer
)